runtime/doc/lsp.txt

```diff
@@ -134,7 +134,8 @@ following (in increasing priority):
 3. Configurations defined anywhere else.
 
 Note: The merge semantics of configurations follow the behaviour of
-|vim.tbl_deep_extend()|.
+|vim.tbl_deep_extend()|. To allow fields to be overriden, fields that have a
+non-literal default are allowed to be set to `false`.
```

Comment:
It might be helpful to include an example here. It's not entirely clear to me what's meant with "non-literal default".

Reply:
A default like `1000` is something a would call a literal default, since the default is a literal value.

`reuse_client` also has a default, however we provide no handle to it, we just provide some abstract text on what the default does.

`name` also has a default of "client id", which again isn't a literal and is different for every server.

Any field that is a list in practice has a literal default of `{}` (empty list), though the docs aren't clear about this.

I'll try and improve the text here, "non-literal default" was the best I could come up with. I welcome any suggestions.

Reply:
I see, thank you for explaining. I think that an example of a setting with a literal default and one that doesn't would help make the term clearer.